src/v/pandaproxy/server.h

```diff
@@ -190,14 +190,15 @@ class auth_ctx_server : public ctx_server<service_t> {
         auto dispatch(Func&& func) {
             switch (authn_method) {
             case config::rest_authn_method::none: {
-                return std::forward<Func>(func)(service().client().local());
+                return std::invoke(
+                  std::forward<Func>(func), service().client().local());
             }
             case config::rest_authn_method::http_basic: {
                 return dispatch([this, func{std::forward<Func>(func)}](
                                   kafka_client_cache& cache) mutable {
                     auto client = cache.fetch_or_insert(user, authn_method);
-                    return std::forward<Func>(func)(*client).finally(
-                      [client] {});
+                    return std::invoke(std::forward<Func>(func), *client)
```

Comment:
Clarification Q: What is the benefit of `std::invoke` here?

Reply:
It's just cleanup, really. It's a generalisation of calling callables, and allows member function pointers, pointers, references, etc.  E.g., `func` could be a member function pointer to `client`.